test/common/verify_alert_links_to_SOPs.go

```diff
@@ -0,0 +1,179 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+)
+
+var githubToken = os.Getenv("TOKEN_GITHUB")
+var failedSOPurls = make(chan string)
+var wg sync.WaitGroup
+var countFailedLinks int
+
+func TestSOPUrls(t TestingTB, ctx *TestingContext) {
+
+	if githubToken == "" {
+		t.Skip("Github token not provided, use GITHUB_TOKEN environment variable to specify it")
+	}
+
+	var sopUrls []string
+
+	// test connection to Github API, with single url
+
+	testUrl := "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/README.md"
+	apiUrl := modifyLink(testUrl)
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", apiUrl, nil)
+	if err != nil {
+		log.Fatal("%s", err)
+	}
+
+	req.Header.Add("Accept", `application/json`)
+	req.Header.Add("Authorization", fmt.Sprintf("token %s", githubToken))
+	testResp, err := client.Do(req)
+	if err != nil {
+		t.Log(err)
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+
+		}
+	}(testResp.Body)
+
+	if testResp.StatusCode != 200 {
+		t.Fatal("Response status: ", testUrl, testResp.Status, "Given token does not allow access to SOP URLs")
+	}
+
+	output, err := execToPod("wget -qO - localhost:9090/api/v1/rules",
+		"prometheus-prometheus-0",
+		ObservabilityProductNamespace,
+		"prometheus", ctx)
+	if err != nil {
+		t.Fatal("failed to exec to pod:", err)
+	}
+
+	var ApiOutput prometheusAPIResponse
+
+	err = json.Unmarshal([]byte(output), &ApiOutput)
```

Comment:
is this a redundant type conversion? :thinking: just a "cosmetic" comment

Reply:
I'm not sure. I have an idea where your comment comming from, but can you please elaborate in more details or suggest me better way?

Reply:
I am not sure myself. But I assume @Stellar-9 just copy-pasted it from e.g. https://github.com/integr8ly/integreatly-operator/blob/master/test/common/alerts_exist.go#L1088-L1098 so maybe we can keep it there for the sake of consistency?
Of course you can try without that (I assume @makslion meant that `byte`) conversion and if it indeed works you can get rid of it.

Reply:
I don't mind either way